docs/migration.md

```diff
@@ -0,0 +1,26 @@
+# Migration guides
+
+## Generic migration
+Migration for the typical mod which doesn't usy anything special but minecraft forge and some library dependencies. 
+For core plugin, mixins, shadowing, access transformes, asm or etc. you'll need to do some extra steps. 
+If they are missing in this document - we will gladly receive your suggestions/contribution. 
+
+1. Copy and replace all files from template to your repository, but: // TODO: (maybe we should have a task in CI which attaches zip with all files but these, called "migration-bundle.zip" or smth)
```

Comment:
maybe we should have a task in CI which attaches zip with all files but these, called "migration-bundle.zip", so it's just "copy and replace all files to your project's root"